code/modules/mob/living/silicon/ai/freelook/chunk.dm

```diff
@@ -1,34 +1,49 @@
-#define UPDATE_BUFFER 25 // 2.5 seconds
+#define UPDATE_BUFFER_TIME 25
 
 // CAMERA CHUNK
 //
 // A 16x16 grid of the map with a list of turfs that can be seen, are visible and are dimmed.
 // Allows the AI Eye to stream these chunks and know what it can and cannot see.
 
 /datum/camerachunk
+	///turfs our cameras cant see but are inside our grid
 	var/list/obscuredTurfs = list()
+	///turfs our cameras can see inside our grid
 	var/list/visibleTurfs = list()
+	///cameras that can see into our grid
 	var/list/cameras = list()
+	///list of all turfs
 	var/list/turfs = list()
+	///camera mobs that can see turfs in our grid
 	var/list/seenby = list()
-	var/changed = 0
+	var/changed = FALSE
 	var/x = 0
 	var/y = 0
 	var/z = 0
 
-// Add an AI eye to the chunk, then update if changed.
-
+/// Add an AI eye to the chunk, then update if changed.
 /datum/camerachunk/proc/add(mob/camera/ai_eye/eye)
 	eye.visibleCameraChunks += src
 	seenby += eye
 	if(changed)
 		update()
+	else
+		var/list/vis_contents_objects = GLOB.cameranet.vis_contents_objects
+		for(var/turf/obscured_turf as anything in obscuredTurfs)
+			obscured_turf.vis_contents += vis_contents_objects
```

Comment:
Why the else here? I assume I'm missing something, but I'd like to make sure I understand this

Reply:
it was a mistake, update() would only add the static object to the vis_contents of turfs no longer visible but were last update(), now it always sets vis_contents |= vis_contents_objects so theres never a repeat